### PR TITLE
issue-#25675 Added fix for #25675 issue to the 2.4 Magento version

### PR DESCRIPTION
**Pull request by sichensol**
Description
This pull request should fix issue #25675 in Magento 2.4

Fixed Issues
#25675: Unable add product to cart in Magento 2.3.3 backend when stock quantity is 1 - "The requested qty is not available"
Manual testing scenarios
Described here #25675

### CODE REVIEW

**Comment by molneek** [comment]
@dmytro-ch 
I see that some tests are failed. E.g ReorderUsingVaultBraintreeTestVariation1_0, but I don't see the place where my code fix or my unit test influence on these test fails. Could you please advise, how to prevent failing the issues and finally add this of the issue to Magento?

**Comment by rogyar** [comment]
Hi @molneek. The `ReorderUsingVaultBraintreeTestVariation1_0` is a fluky test. It will be fixed soon after merging a corresponding fix into mainline. So, don't bother with this failing test, it's not related to your change somehow

**Comment by molneek** [comment]
Hi @rogyar. So in this case, does it mean that all these tests "Functional Tests B2B, Functional Tests CE, Functional Tests EE and Static Tests" are not related to my changes? 
Do I need manually rerun check when these are fixed, or this happen automatically?

**Comment by rogyar** [comment]
@magento run all tests

**Comment by molneek** [comment]
Hello @rogyar, seems like all tests passed

File: app/code/Magento/CatalogInventory/Test/Unit/Model/Quote/Item/QuantityValidator/QuoteItemQtyListTest.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+{
+    /**
+     * @var QuoteItemQtyList
+     */
+    private $quoteItemQtyList;
+
+    /**
+     * @var int
+     */
+    private $itemQtyTestValue;
```
Comment (rogyar) [inline app/code/Magento/CatalogInventory/Test/Unit/Model/Quote/Item/QuantityValidator/QuoteItemQtyListTest.php:26]: Just a hint for the future. If we don't need to share value between different methods, there's no need to create an additional class property. You may use just local variables inside of the methods.

So `$this->itemQtyTestValue` will be simply `$itemQtyTestValue` inside of every method (test) of the current class

**Review by rogyar — CHANGES_REQUESTED** [review]

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-6910](https://jira.corp.magento.com/browse/ENGCOM-6910) has been created to process this Pull Request
:eight_spoked_asterisk: @rogyar, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

An order could be created and proceed without any errors if the product quantity is **1**

File: app/code/Magento/CatalogInventory/Test/Unit/Model/Quote/Item/QuantityValidator/QuoteItemQtyListTest.php (hunk 1 above)
Comment (sichensol) [reply app/code/Magento/CatalogInventory/Test/Unit/Model/Quote/Item/QuantityValidator/QuoteItemQtyListTest.php:26]: @rogyar Thank you, I agree, I had some ideas about using it, but then changed my mind and just used this as this is already created

**Pull request merged**